include/TS3.h

```diff
@@ -3,27 +3,36 @@
 
 #include <iostream>
 
-#include "TDetector.h"
+#include "TGRSIDetector.h"
 #include "TS3Hit.h"
 
-class TS3 : public TDetector {
+class TS3 : public TGRSIDetector {
 	public:
 		TS3();
-		~TS3();
+		TS3(const TS3&);
+		virtual  ~TS3();
 
-		void AddFragment(TFragment*, MNEMONIC*);
-		void BuildHits() {} //no need to build any hits, everything already done in AddFragment
+		virtual void AddFragment(TFragment*, MNEMONIC*);
+		virtual void BuildHits();
 
+		TGRSIDetectorHit* GetHit(const int& idx =0);
 		TS3Hit* GetS3Hit(const int& i);  
-		Short_t GetMultiplicity() { return fS3Hits.size(); }
+		Short_t GetMultiplicity() const { return fS3Hits.size(); }
+		void PushBackHit(TGRSIDetectorHit* deshit);
 
-		TVector3 GetPosition(int front, int back);
+		static TVector3 GetPosition(int ring, int sector);
 
-      virtual void Clear(Option_t *opt = "all");		     //!
-      virtual void Print(Option_t *opt = "") const;		  //!
+		void Copy(TObject&) const;
+		TS3& operator=(const TS3&);  // 
+		virtual void Clear(Option_t *opt = "all");		     //!
+		virtual void Print(Option_t *opt = "") const;		  //!
+		
+		
 
 	private:
 		std::vector<TS3Hit> fS3Hits;
+		std::vector<TFragment*> fS3_RingFragment; //! 
+		std::vector<TFragment*> fS3_SectorFragment; //! 
```

Comment:
Understood, I would still recommend you take a look at the build hits
method of csm.  It been a bit warpped over time but it shows you can
determine the order of front back coincidence at build time.  you can
alternatively throw every think in the analysis tree and punt the problem
down the road.   The danger here of course, is anyone not completely aware
of what is happening is very likely to double count events...

On Wed, Nov 4, 2015 at 9:32 PM, jsmallcombe <notifications@github.com>
wrote:

> In include/TS3.h
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584#discussion_r43970276>
> :
>
> >
> >  	private:
> >  		std::vector<TS3Hit> fS3Hits;
> > +		std::vector<TFragment*> fS3_RingFragment; //!
> > +		std::vector<TFragment*> fS3_SectorFragment; //!
>
> And again, we want to identify events for what they are, even though they
> may not be what we want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584/files#r43970276>
> .
>